### PR TITLE
Don't calculate slots when MRO parsing fails.

### LINKED ISSUE #1088: astroid.exceptions.DuplicateBasesError raised when not needed to.
### Steps to reproduce

1. Create a file with contents 
```py
def generate_class():
    class GeneratedClass:
        pass
    return GeneratedClass

class Klass(generate_class(), generate_class()):
    def __init__(self):
        self.attribute = 'value'
```
2. Run pylint (pylint 2.9.3, astroid 2.6.2, Python 3.6.12)
3. Pylint will fail hard with output 
```
Traceback (most recent call last):
  File "astroid/decorators.py", line 36, in cached
    return cache[func]
KeyError: <bound method ClassDef.slots of <ClassDef.Klass l.6 at 0x10bd14358>>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "pylint/lint/pylinter.py", line 1144, in get_ast
    return MANAGER.ast_from_file(filepath, modname, source=True)
  File "astroid/manager.py", line 116, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
  File "astroid/builder.py", line 135, in file_build
    return self._post_build(module, encoding)
  File "astroid/builder.py", line 155, in _post_build
    self.delayed_assattr(delayed)
  File "astroid/builder.py", line 233, in delayed_assattr
    if not _can_assign_attr(inferred, node.attrname):
  File "astroid/builder.py", line 57, in _can_assign_attr
    slots = node.slots()
  File "astroid/decorators.py", line 38, in cached
    cache[func] = result = func(*args, **kwargs)
  File "astroid/scoped_nodes.py", line 2906, in slots
    slots = list(grouped_slots())
  File "astroid/scoped_nodes.py", line 2891, in grouped_slots
    for cls in self.mro()[:-1]:
  File "astroid/scoped_nodes.py", line 2978, in mro
    return self._compute_mro(context=context)
  File "astroid/scoped_nodes.py", line 2966, in _compute_mro
    unmerged_mro = list(clean_duplicates_mro(unmerged_mro, self, context))
  File "astroid/scoped_nodes.py", line 157, in clean_duplicates_mro
    context=context,
astroid.exceptions.DuplicateBasesError: Duplicates found in MROs (Klass), (GeneratedClass, object), (GeneratedClass, object), (GeneratedClass, GeneratedClass) for <ClassDef.Klass l.6 at 0x10bd14358>.
************* Module b
b.py:1:0: F0002: <class 'astroid.exceptions.DuplicateBasesError'>: Duplicates found in MROs (Klass), (GeneratedClass, object), (GeneratedClass, object), (GeneratedClass, GeneratedClass) for <ClassDef.Klass l.6 at 0x10bd14358>. (astroid-error)```

### Current behavior
Pylint fails parsing the file.

### Expected behavior
Pylint should be able to parse the file

### 2.6.2

### PR DESCRIPTION
**Pull request by doranid**
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description

`slots()` won't now fail if it can't parse the mro object.
I'm not sure that's the right way to fix that.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Related Issue

#1088 

Thanks!

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
Probably the astroid fix we need for https://github.com/PyCQA/pylint/pull/4688, thank you !

**Comment by doranid** [comment]
> Probably the astroid fix we need for [PyCQA/pylint#4688](https://github.com/PyCQA/pylint/pull/4688), thank you !

These two issues are orthogonal as per my understanding.

Thanks!

File: astroid/scoped_nodes.py
Comment (Pierre-Sassoulas) [inline astroid/scoped_nodes.py:2889]:
```suggestion
        def grouped_slots(mro: ?) -> None:
```
Could you add typing on modified functions, please ? :)

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
I don't know how I would test that, but would it be possible to add an automated test for this change ?

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank for adding a test, I'm going to wait for another reviewer before merging :)

**Pull request merged**
